In [6]:
import pandas as pd
import numpy as np
# loading the mercari data
df=pd.read_csv('C:\\Users\\layal\\OneDrive\\Documents\\Infinite\\LAU\\DataScience\\AAI614_Zeineddine\\Week_3\\train.tsv',sep='\t')

def detect_outliers_iqr(data_column):
    # Sort data is handled by pandas quantile
    Q1 = data_column.quantile(0.25)
    Q3 = data_column.quantile(0.75)
    IQR = Q3 - Q1
    
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR
    
    # Identify outliers
    outliers = data_column[(data_column < lower_bound) | (data_column > upper_bound)]
    
    return outliers, lower_bound, upper_bound


outliers, lb, ub = detect_outliers_iqr(df['price'])

print(f"Lower Bound: {lb}")
print(f"Upper Bound: {ub}")
print(f"Number of outliers detected: {len(outliers)}")

Lower Bound: -18.5
Upper Bound: 57.5
Number of outliers detected: 119352


In [14]:
# Look at the most expensive 'outliers'
print("Top 5 most expensive outliers:")
print(df[df['price'] > 57.5].sort_values(by='price', ascending=False).head())

Top 5 most expensive outliers:
         train_id                                      name  \
760469     760469       NEW Chanel WOC Caviar Gold Hardware   
1262245   1262245  NEW-Chanel Boy Wallet o Chain WOC Caviar   
1393600   1393600                 David Yurman Wheaton ring   
778940     778940                                  Mary kay   
1445348   1445348         Vs2/I 1ct Diamond Engagement Ring   

         item_condition_id                                 category_name  \
760469                   1           Women/Women's Handbags/Shoulder Bag   
1262245                  1  Women/Women's Handbags/Messenger & Crossbody   
1393600                  2                           Women/Jewelry/Rings   
778940                   1                            Beauty/Makeup/Face   
1445348                  2                           Women/Jewelry/Rings   

           brand_name   price  shipping  \
760469         Chanel  2009.0         0   
1262245        Chanel  2006.0         0   
1393

$The IQR analysis flagged items priced above $57.50 as outliers, which at first glance could be interpreted as noise or data quality issues. However, looking more closely at these records shows that many of them are perfectly valid within their business context. For example, items such as Chanel handbags and diamond engagement rings naturally appear at much higher price points and should not be treated as errors simply because they fall outside a statistical threshold. In another case, a $2,000 Mary Kay listing initially looks abnormal, but the item description reveals it represents a bulk sale rather than a single product.

These examples also highlight several data quality concepts discussed in Week 3. The Mary Kay listing represents a pattern violation relative to typical makeup prices, even though the value itself is valid. The diamond ring entry illustrates incompleteness, as the brand name is missing despite the high price of the item. In addition, all of these high-priced items have low item_condition_id values (1 or 2), reinforcing the strong relationship between item condition and price.

Overall, this analysis shows that the detected outliers are genuine extremes rather than random noise. Removing them would result in the loss of meaningful and potentially important data. A more appropriate approach for future modeling would be to keep these observations and apply techniques such as log transformation to handle the wide range of prices.$